src/librbd/ImageWatcher.cc

```diff
@@ -1135,14 +1172,28 @@ bool ImageWatcher<I>::handle_payload(const SnapRenamePayload &payload,
     int r;
     if (m_image_ctx.exclusive_lock->accept_request(
           exclusive_lock::OPERATION_REQUEST_TYPE_GENERAL, &r)) {
-      ldout(m_image_ctx.cct, 10) << this << " remote snap_rename request: "
-			         << payload.snap_id << " to "
-			         << payload.snap_name << dendl;
-
-      m_image_ctx.operations->execute_snap_rename(payload.snap_id,
-                                                  payload.snap_name,
-                                                  new C_ResponseMessage(ack_ctx));
-      return false;
+      bool new_request;
```

Comment:
Nit: it would be great if we could get a helper function to handle all of this boilerplate code